hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -4568,8 +4576,29 @@ public boolean isAffectedByEnabledFetchProfiles(LoadQueryInfluencers loadQueryIn
 
 	@Override
 	public boolean isAffectedByEnabledFilters(LoadQueryInfluencers loadQueryInfluencers) {
-		return loadQueryInfluencers.hasEnabledFilters()
-				&& filterHelper.isAffectedBy( loadQueryInfluencers.getEnabledFilters() );
+		if ( loadQueryInfluencers.hasEnabledFilters() ) {
+			if ( filterHelper.isAffectedBy( loadQueryInfluencers.getEnabledFilters() ) ) {
+				return true;
+			}
+			// we still need to verify collection fields to be eagerly loaded by 'join'
+			final NonIdentifierAttribute[] attributes = entityMetamodel.getProperties();
+			for ( NonIdentifierAttribute attribute : attributes ) {
+				if ( attribute instanceof EntityBasedAssociationAttribute ) {
+					final AssociationType associationType = ( (EntityBasedAssociationAttribute) attribute ).getType();
+					if ( associationType instanceof CollectionType ) {
+						final Joinable joinable = associationType.getAssociatedJoinable( getFactory() );
+						if ( joinable.isCollection() ) {
+							final QueryableCollection collectionPersister = (QueryableCollection) joinable;
+							if ( collectionPersister.getFetchMode() == FetchMode.JOIN
+									&& collectionPersister.isAffectedByEnabledFilters( loadQueryInfluencers ) ) {
+								return true;
+							}
+						}
+					}
+				}
+			}
+		}
+		return false;
```

Comment:
The second bug flushed out. The original implementation doesn't take into consideration the following edge case:
1. enabled filter is not for the owner entity;
2. there is a collection field to be loaded eagerly by `join`;
3. enabled filter is for this collection's element entity;